nemo_curator/utils/semdedup_utils.py

```diff
@@ -379,148 +201,48 @@ def prune_single_cluster(
     Args:
         cluster_id (int): The specific cluster ID to process.
         id_col (str): The name of the ID column.
-        id_col_type (str): The data type of the ID column.
-        sorted_clusters_dir (str): Path to the sorted clusters directory.
+        emb_by_clust_dir (str): Path to where clustered embeddings are stored.
         semdedup_pruning_tables_dir (str): Path to the pruning tables directory.
         eps (float): Epsilon value for pruning.
 
     Returns:
         cudf.DataFrame: A DataFrame of the pruned cluster data
     """
-    sorted_fname = os.path.join(sorted_clusters_dir, f"cluster_{cluster_id}.npy")
-    if not os.path.exists(sorted_fname):
-        return _get_empty_results_df(id_col, id_col_type)
-
-    # Read the sorted cluster file
-    # Once we change sorted file to parquet we can just read the DF here instead of loading the npy file and converting to DF
-    cluster_data = np.load(sorted_fname)
-    df_cluster = cudf.DataFrame(
-        {
-            id_col: cluster_data[:, 0],
-            "dist": cluster_data[:, 1],
-            "cluster": cluster_data[:, 2],
-        }
-    )
+    cluster_dir = os.path.join(emb_by_clust_dir, f"nearest_cent={cluster_id}")
+    df_cluster = cudf.read_parquet(
+        cluster_dir, columns=[id_col, COSINE_DIST_TO_CENT_COL]
+    ).assign(cluster=cluster_id)
 
-    df_cluster[id_col] = df_cluster[id_col].astype(id_col_type)
-    df_cluster["dist"] = df_cluster["dist"].astype("float32")
-    df_cluster["cluster"] = df_cluster["cluster"].astype("int32")
-
-    # TODO : we don't need to read the pruning table here as we can just filter on the cosine_sim_score in the sorted cluster file once we add it
-    # Read the pruning table
     pruning_table_fname = os.path.join(
         semdedup_pruning_tables_dir, f"cluster_{cluster_id}.parquet"
     )
-    pruning_table = cudf.read_parquet(pruning_table_fname)
+    # TODO should we add max_id to for the user
```

Comment:
I don't understand the purpose of this TODO?

Reply:
Changed the comment to
`# In future we can add more columns to the pruning table like max_id etc.`